src/DoubleLinkedList.sol

```diff
@@ -79,12 +77,11 @@ library DoubleLinkedList {
     /// @param id The address of the account.
     function remove(List storage list, address id) internal {
```

Comment:
this function can be optimized:

```solidity
function remove(List storage list, address id) internal {
    Account storage account = list.accounts[id];
    if (id == address(0)) revert AddressIsZero();
    if (account.value == 0) revert AccountDoesNotExist();

    address prevId = account.prev;
    address nextId = account.next;
    list.accounts[prevId].next = nextId;
    list.accounts[nextId].prev = prevId;

    account.prev = address(0);
    account.next = address(0);
    account.value = 0;
}
```
<details>
<summary>gas diff</summary>
<img src="https://user-images.githubusercontent.com/108467407/237045289-e6dfe22b-a0ef-4674-a7b6-6e3d9122afe6.png">
</details>


The line `if (id == address(0)) revert AddressIsZero();` can be removed, but the error thrown would no more be accurate in case of null address.

Reply:
I will implement but not sure we will merge the PR as mentioned above